lib/active_model/serializer.rb

```diff
@@ -238,10 +238,9 @@ def json_key
 
     # Return the +attributes+ of +object+ as presented
     # by the serializer.
-    def attributes
-      attributes = self.class._attributes.dup
-
-      attributes.each_with_object({}) do |name, hash|
+    def attributes(requested_attrs = nil)
+      self.class._attributes.each_with_object({}) do |name, hash|
+        next unless requested_attrs.nil? || requested_attrs.include?(name)
```

Comment:
why should `request_attrs` default to nil? Shouldn't it be an array?

and why not just assume they're giving valid attribute names? or compare the two lists?

```ruby
def attributes(attribute_names = self.class._attributes)
  attribute_names.each_with_object({}) do |name, hash|
```

Also, this might be a good spot to pass in `options[:fields]` as `{only: options[:fields]}` so that we can also support `except`. Shrug.